### ISSUE TITLE
CRAN task view proposal: DataQualityInHealth

### ISSUE THREAD

**Issue by JoanyMarino** [cran-task-views/ctv#28]
# Scope

*DataQualityInHealth* contains packages that help assess data quality, focusing on phenotypic data from health and social science studies.

Data quality is the degree to which data fulfills defined requirements, and it is an issue in all scientific fields. Evaluating data quality is essential in initial data analysis and quality monitoring processes to ensure valid research results. Base R already contains general functionalities for assessing data quality in a broad sense, and depending on the application, users may employ several packages. Yet, users will commonly need extensive programming to create comprehensive reports, with code being typically highly tailored to a specific data set. Therefore, it is important to identify packages with particular functionalities to facilitate conducting data quality analyses. Using these packages may also improve the comparable reporting on data quality across studies. However, the applicability of any such package, together with suitable data quality approaches, will differ across science fields as determined by the data structures, its requirements, and the representation of the latter in the metadata. This task view documents packages for data quality assessment, specifically for phenotypic data with a focus on health and social science studies. Prototypical data examples are results from clinical measurements (e.g., blood pressure) or responses to surveys and questionnaires. Nevertheless, this focus does not preclude the suitability of the documented packages for use in other research fields.

The quality checks performed by the targeted packages include inspecting a range of data quality dimensions, such as integrity (the degree to which the data conforms to structural and technical requirements), completeness (the degree to which expected data values are present), consistency (the degree to which data values are free of breaks in conventions or contradictions), and accuracy (the degree of agreement between observed and expected distributions and associations). The packages are broadly divided according to their capabilities into three major groups: packages that explicitly focus on data quality assessments (these usually combine targeted data quality checks with descriptive or explorative features), packages that target data exploration and statistical analyses but provide data quality-related functions, and packages that perform focused data quality checks based on an input of rules and requirements. Many have functionalities that span more than one category. We include packages, if they fall (at least) into one of the three categories mentioned above, are applicable to spreadsheet type data as commonly used in phenotypic data collected in the health data in various formats, and provide a sufficient number of data quality checks. For the latter, we compared packages in reference to a data quality framework for observational health research data ([Schmidt et al. 2021](https://doi.org/10.1186/s12874-021-01252-7)). As a threshold for inclusion, the packages need to address at least three data quality dimensions and four domains of the reference framework. Packages that do not run as expected or produce major errors when applied to evaluate 
[publicly available data sets](https://dataquality.ship-med.uni-greifswald.de/ExampleDataDescription.html) are not included.

# Packages 

Please see [this document](https://github.com/JoanyMarino/DataQualityInHealth/blob/main/DataQualityInHealth.md) for the list of packages and a preview of the Task View. Currently, it consists of 27 packages (including 9 core packages).  
  
# Overlap 

Data quality assessment and monitoring are included in other task views only to a smaller extent. Some views list specific applications, for example, hydrological data quality in the [Hydrology](https://CRAN.R-project.org/view=Hydrology) task view. The [MissingData](https://CRAN.R-project.org/view=MissingData) task view reviews data completeness and imputation in detail, which we consider as part of a data quality assessment. Similarly, quality inspection during general data processing is addressed in the [OfficialStatistics](https://CRAN.R-project.org/view=OfficialStatistics) task view but limited to assertion checking packages. We aim to minimize overlap with these broad task views by concentrating on observational health data. A potential overlap with a focus on medical data would be the [ClinicalTrials](https://CRAN.R-project.org/view=ClinicalTrials); however, [ClinicalTrials](https://CRAN.R-project.org/view=ClinicalTrials) places a greater emphasis on the design and analysis of clinical trial data, while the packages that are included in DataQualityInHealth are meant to be used at a different point in the research pipeline, namely, during ongoing data collections or when the data becomes available before the start of substantive statistical analyses. 
  
# Maintainers 

The primary co-maintainers are Elisa Kasbohm (@ekasb) and Joany Mariño (@JoanyMarino). Co-maintainers include: Stephan Struckmann (@struckma), Lorenz A. Kapsner (@kapsner), and Carsten O. Schmidt (@c3schmidt).

**Comment by davidjohannesmeyer** [comment]
This is a very interesting task view, obviously of general interest. Actually, in its current state, only very few packages seem to be limited to or focused on Health/Social Science data. Personally, I would not limit the scope to these two application domains and just use the name "Data Quality", maybe flagging out the specialized packages, or adding separate sections if there are more of them.

**Comment by zeileis** [comment]
I agree with David @davidjohannesmeyer that the general topic would be of interest but that the current scope is not appropriate. I understand that this is the core interest of the group who proposed this task view and well-aligned with their [DFG/EU research project](https://dataquality.ship-med.uni-greifswald.de/). But the scope is too limited for a CRAN task view and all but one of the maintainers come from the same institution which is also not likely to generate a broad and diverse overview.

Hence I think that this proposal cannot be considered directly for CRAN. I'll leave this GitHub issue open, though, so that my fellow CRAN Task View Editors (@rsbivand @eddelbuettel @rociojoo @tuxette) as well as other interested users can agree or disagree with me. Maybe @matthias-da has some thoughts, too.

In case one, or maybe two, of the authors of this proposal are willing to do the additional work, I think it might be of interest to work on a new and more general data quality task view proposal. I'll provide some further thoughts on this in a separate comment.

**Comment by zeileis** [comment]
**Feedback for a general data quality and data cleaning task view:**

**Inclusion/exclusion criteria:** These have to be carefully defined because there are quite a few related topics that would make the task view too broad.
* _Data wrangling:_ There are lots of general-purpose as well as specialized packages for this in R. A distinction has to be found which of these packages is really dedicated to data quality or data cleaning.
* _Exploratory data analysis:_ Similar, there are lots of packages that generate exploratory overviews of data. But when are these sufficiently focused on data quality?
* _Reporting:_ Finally, there are packages that generate overview tables of data along with infrastructure to embed them in reports. This includes several packages for "Table 1" generation for scientific papers or general reporting infrastructure. The [ReproducibleResearch](https://CRAN.R-project.org/view=ReproducibleResearch) task view has some related discussion. I would expect that most of this is not eligible for a data quality task view but again it would be good to make clear what constitutes the distinction.

For a general data quality and data cleaning task view it would be important to clearly define the inclusion and exclusion criteria. In particular it needs to be clear that general "data wrangling" tools are not within scope because that would be far too broad. Similarly, some distinction has to be made which tools for "exploratory data analysis" are

**Related task views:** It would be important to properly connect to related task views and guide readers how these complement the discussion. Obvious candidates are [MissingData](https://CRAN.R-project.org/view=MissingData) and [OfficialStatistics](https://CRAN.R-project.org/view=OfficialStatistics), possibly also [ReproducibleResearch](https://CRAN.R-project.org/view=ReproducibleResearch) and [Databases](https://CRAN.R-project.org/view=Databases). Additionally, some connections might be made to specific disciplines, e.g., a proposal for an [Epidemiology task view](https://github.com/cran-task-views/ctv/issues/25) is currently under preparation.

**Maintenance:** The principal maintainer for the task view should be a person that is willing to commit to this voluntary work for a longer time. It's really not just about writing the initial version but doing all the subsequent maintenance work over years. This includes regularly incorporating new (CRAN) packages, responding to user suggestions, communicating with the CRAN Task View Editors, making other updates, e.g., when packages are archived on CRAN etc. Of course, the co-maintainers can help with all of these task but the principal maintainer is often the first point of contact. (Note also that the principal maintainer must be a single person and there are no principal co-maintainers.) Ideally, a diverse group of co-maintainers is formed where diversity can be in terms of gender, ethnicity, academic field, geographic location, etc.

To form such a diverse group I would recommend that one (or maybe two) of the persons from the current proposal reach out to other potential co-maintainers. One natural idea would be to ask one of the authors of the `validate` package who have also authored a corresponding [book](https://doi.org/10.1002/9781118897126) and [scientific paper](https://doi.org/10.18637/jss.v097.i10) about it. Possibly some of the other authors/maintainers from recommended packages might add to the diversity as well, I haven't checked this in detail, yet.

**Comment by rsbivand** [comment]
Agree with @zeileis . As with my comment on another proposal https://github.com/cran-task-views/ctv/issues/27#issuecomment-1131901329, it is vital that task views are broad enough to merit continuing support if the initiative-takers retire. Task Views involve much more ongoing commitment than packages. Yes, a broad, general data quality task view might be worth considering; I would also value the input of the authors of the **validate** package.

**Comment by matthias-da** [comment]
In my view, limiting it to health observational data is too much of a restriction even when it is extended to important topics  like data fushion and record linkage, health data and their quality after anonymization, selection bias and methods to reduce bias, and with a stronger focus on missing data (imputation and calibration) and outliers (robust statistics).   

However, a general data quality task will raise many difficulties. 
A general task view on data quality will have large overlaps with existing task views. We would produce a lot of overlap with existing task reviews and it would be a lot of work to harmonize them, if at all possible. 

BTW, for [OfficialStatistics](https://cran.r-project.org/web/views/OfficialStatistics.html), for example, we follow the data quality scheme [GSBPM](https://statswiki.unece.org/display/GSBPM) and I would avoid putting all of this in a separate data quality task view because these reported data quality issues are so important to official statistics and survey statistics. In other words, one of the biggest parts in official statistics is data quality, and without the data quality issues reported in this task view, the official statistics task view would remain naked. Just wanted to mention this to see the difficulties to create an own general data quality task view.

**Comment by zeileis** [comment]
Matthias @matthias-da, thanks for your feedback! I agree that we have to be careful with overlap between `OfficialStatistics` and a new task view. Why I thought that a new separate task view might be viable was that the list of packages in the proposal has very little overlap with the list of packages in `OfficialStatistics`. So one possibility would be to separate the views similar to what you have done with the `MissingData` task view (which also wasn't too easy). Or do you think that everything should go into `OfficialStatistics`?

**Comment by JoanyMarino** [comment]
Thanks to all for the valuable feedback!

Before deciding the way forward, we want to continue the discussion by considering two possibilities:
 
 1. Define a more general data quality task view (however, without very context-specific data quality aspects, such as statistical power). While there is some overlap with the areas mentioned above (such as exploratory data analysis, data wrangling, and reporting), data quality is set apart significantly by referring to the degree to which data fulfills defined requirements. Therefore, packages that qualify should be able to extensively perform explicit rule-based checks of data properties against specified conditions. We would define new inclusion/exclusion criteria and invite a diverse group of co-maintainers. 

2. Retain the focus on health observational studies by integrating data quality as another section in the [epidemiology task view](https://github.com/cran-task-views/ctv/issues/25). What do you think @[thibautjombart](https://github.com/thibautjombart), [@Bisaloo](https://github.com/Bisaloo) and [@mjrolland](https://github.com/mjrolland)?
 
Other thoughts: 

The inclusion of data quality in [Official Statistics](https://cran.r-project.org/web/views/OfficialStatistics.html) may not be useful from a user viewpoint. Sure there is overlap. However, data quality is much wider than "Official Statistics & Survey Statistics". Some packages mentioned in [Official Statistics](https://cran.r-project.org/web/views/OfficialStatistics.html) are not bound to be used in "official statistics and survey statistics", such as `validate`.

**Comment by matthias-da** [comment]
Thanks to all comments, it might get clearer and clearer, at least from my perspective ;-)

Since data quality is not only about reporting capabilities, but also about collection, missing values, outliers and erroneous values, consistency and additivity of results, etc., Therefore, the maintainers of the [Official Statistics](https://cran.r-project.org/web/views/OfficialStatistics.html) think that a data quality task will cut too much from other existing CTVs.

**Some reasons for not having a task view on data quality:**
Generally, a data quality CTV would be very generic and would have to touch all areas of statistics. 
Examples: If especially the editing/validation packages written by national statistical offices were not included in the CTV of OfficialStatistics, we would find it strange. Likely this is also true for task views on, for example, missingData when they should cut their parts on visual inspection of missing values because it is a data quality problem. And the task view on robust statistical methods would need to cut methods because the topic of outliers is also a major data quality issue. There might be dozens of other examples in different areas.

**How to integrate data quality aspects to exising CTVs:**
The OfficialStatistics CTV might be extended with those aspects which concern the data quality in surveys best (so far they are missing). If possible, each existing CTV may specifically mention data quality aspects in their area and if it does not concern data quality in surveys (where one could otherwise refer), instead of making a generic CTV about data quality. 

So, personally I would prefer (2) from @[JoanyMarino](https://github.com/JoanyMarino) .

Small note: `validate` was written by Statistics Netherlands, so it is really intentionally designed for surveys and populations from official statistics. And yes, it might be useful for others outside this community, but the package is a very central package for [Official Statistics](https://cran.r-project.org/web/views/OfficialStatistics.html). It would be just strange to not discuss it in [Official Statistics](https://cran.r-project.org/web/views/OfficialStatistics.html) (to avoid overlap with a generic data quality CTV).

**Comment by davidjohannesmeyer** [comment]
The idea of a TV is to give some guidance to *unexperienced* users. These might not expect important packages on data quality in "Official Statistics", for example.

We might consider a short TV on Data Quality that mainly gives an overview and references Official Statistics, MissingData etc. to avoid duplicating package lists and maintenance efforts, but allows for listing packages not mentioned elsewhere. So sort of a meta-TV (at least partly).

David

**Comment by zeileis** [comment]
@matthias-da thanks for your thoughts. I agree that the issue of potential overlap has to be considered - and the inclusion/exclusion criteria have to be explained in order to keep things manageable. But I'm not quite as pessimistic as you regarding the feasibility. Also, I think you are exaggerating the necessity of cutting something from other task views. For example:

- The `MissingData` is a relatively focused task view so I would just list a couple of the most important packages in a `DataQuality` task view as well and otherwise refer to the `MissingData` task view. This is essentially what the `OfficialStatistics` task view is also doing now.
- Similarly I would expect that it shouldn't be too hard to just mention the crucial robust statistics aspects and then refer to the `Robust` task view.
- And while we try to avoid overlap between the task views, it is certainly ok to have important packages listed in several views. For example, `validate` is certainly important enough to have it listed in both views.

At least it's my impression that this would be easier to do than to split up the content of the proposed `DataQualityInHealth` task view. In that case, the health-specific stuff would have to go into the forthcoming `Epidemiology` task view and everything else into the `OfficialStatistics` task view. But both of these task views are already quite large.

What do the other CRAN Task View Editors think? Should we encourage the proposal of a general `DataQuality` task view? Or rather split things up? Or do something else? @rsbivand @eddelbuettel @rociojoo @tuxette

**Comment by eddelbuettel** [comment]
I second the more relaxed view by @zeileis.  We had overlap between packages and views, and 'multiple assocations' of packages to more than one view -- that's just how it goes.

**Comment by c3schmidt** [comment]
Dear all,

thanks for the very interest and helpful comments. Here are my 5cents:

The current CTVs are very different in their conceptual focus. Some, like [MissingData](https://cran.r-project.org/web/views/MissingData.html) , [MetaAnalysis](https://cran.r-project.org/web/views/MetaAnalysis.html), [Survival](https://cran.r-project.org/web/views/Survival.html) have a clear methodological focus.

Others, like [OfficialStatistics](https://cran.r-project.org/web/views/OfficialStatistics.html), or the proposed Epidemiology view cover a wide range of functionalities of relevance in some field of research. Many of the methodological aspects mentioned within these are not unique to that particular area of application as for example in [OfficialStatistics](https://cran.r-project.org/web/views/OfficialStatistics.html), e.g. missing data , outlier detection and appear with good reason elsewhere. 

From a didactic pint of view, as mentioned by [eddelbuettel](https://github.com/eddelbuettel) , the heterogeneous conception of task views implicates a useful multiple mentioning of single packages to ensure finding them either by field of research or methodological area of interest. In and by itself mentioning a package only once is of no value. It should be about finding what the user desires in his/her work flow. (BTW: It might be useful to group CTVs by either of the two major approaches they take in including packages)

So turning to the data quality issue. When taking the arguments above into account and looking at the granularity of CTVs,  a DataQualityInHealth CTV would not be desirable. Why? It mixes up the methodological aspect "data quality" with a specific field of research "observational health research". 

What might be solutions:

- When retaining a focus on "observational health research" it could simply be a new subsection in a the proposed Epidemiology CTV . (In only fear that due to the enormous complexity of Epidemiology, if that view gets seriously maintained, covering what Epidemiology really is, it will get crazy long -btw thats why I said in TV proposal  [25 ](https://github.com/cran-task-views/ctv/issues/25)that something like Infectious diseases fits better and is in granularity much more like OfficialStatistics, an epidemiology CTV would also strongly overlap with OfficialStatistics because a lot in Epidemiology is about surveys and official statistics)
-  Establish a generic "data quality" task view. In granularity it fits well as data quality checks are a dedicated step in a scientific workflow. Contrary to other statements above, I think it will not take much from any other existing CTV. Sure, aspects like missing values, extreme values, outliers,  form part of data quality with highly specialized packages. But that is not the genes of the intended packages for a CTV data quality Specific packages would just be addressed by linking to existing sections, not duplicating them. In scope would be to me packages that provide an overarching approach on facilitating data quality assessments across a wide range of issues with a considerable simplification in workflows over using a series of specific packages.

**Comment by rsbivand** [comment]
With this iteration, I feel @c3schmidt 's suggestions may be a way forward, but would need agreement from the epidemiology TV proposers https://github.com/cran-task-views/ctv/issues/25 that gentle constraining to infectious diseases works for them. Being able to link between rendered TV documents will here help in the division of labour. However, TV maintainers may need pinging when a section of another TV that they link to is edited (for some definition of a non-trivial edit).

**Comment by rociojoo** [comment]
I agree with the idea of a "data quality" TV if the maintainers think that they could make it general enough. I think it can be very useful for inexperienced users as @davidjohannesmeyer mentioned, and that there won't be as much overlap—that should be checked and sections of CTVs like `Official Statistics` or `Epidemiology` should be cited when necessary. But again, that really depends on whether the maintainers feel like they can work on such a proposal and commit to maintaining it. 
If not, the option of contributing to the `Epidemiology` or the `Official Statistics` TV would be better.

**Comment by c3schmidt** [comment]
Hi [rociojoo](https://github.com/rociojoo) yes, if either of the two solutions finds acceptance we will discuss internally with the group to figure out which scope is manageable and likely implicate others, we'll come back to you, best

**Comment by zeileis** [comment]
I agree. Both routes (new general task view vs. contributions to existing task views) are conceivable. So please go ahead with your internal discussions and follow up here.

**Comment by c3schmidt** [comment]
Ok, thanks for the many comments, we will do so and come back shortly to you,

**Comment by thibautjombart** [comment]
Sorry for jumping in late on this thread, I had missed the gh notifications. I like the idea of adding a section on data quality to the [Epidemiology](https://cran.r-project.org/web/views/Epidemiology.html) TV, and happy to help making this happen.

**Comment by tuxette** [comment]
This issue is inactive for more than 6 months now so I am closing it.
